### PR TITLE
Use readonly distance heap in exact nearest neighbor iterators during

### PR DESCRIPTION
**Pull request by toregge**
later matching phases.

@geirst : please review

### CODE REVIEW

**Review by geirst — DISMISSED** [review]
👍

File: searchlib/src/vespa/searchlib/queryeval/exact_nearest_neighbor_iterator.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace search::queryeval {
  * Keeps a heap of the K best hit distances.
  * Currently always does brute-force scanning, which is very expensive.
  **/
-template <bool strict, bool has_filter, bool has_single_subspace>
+template <bool strict, bool has_filter, bool readonly_distance_heap, bool has_single_subspace>
```
Comment (baldersheim) [inline searchlib/src/vespa/searchlib/queryeval/exact_nearest_neighbor_iterator.cpp:22]: No need to readonly_distance_heap as a template argument. It is only used in unpack, which a very cold method.

**Review by geirst — APPROVED** [review]
👍

**Review by baldersheim — APPROVED** [review]

File: searchlib/src/vespa/searchlib/queryeval/exact_nearest_neighbor_iterator.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/exact_nearest_neighbor_iterator.cpp]: Perfect

**Review by baldersheim — APPROVED** [review]

**Pull request merged**